sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/BlobCheckpointStore/BlobsCheckpointStore.cs

```diff
@@ -301,13 +305,15 @@ await foreach (BlobItem blob in ContainerClient.GetBlobsAsync(traits: BlobTraits
 
                     if (startingPosition.HasValue)
                     {
-                        checkpoints.Add(new EventProcessorCheckpoint
+                        checkpoints.Add(new BlobEventProcessorCheckpoint
```

Comment:
I'm not a fan of adding the additional type here;  I see where you're using it, but I think the metrics approach is worthy of discussion and may obviate this.   If we do end up keeping it, can we consider changing to to reflect the storage mechanism?   Maybe something like `BlobStorageCheckpoint` since the processor itself doesn't know or care about it?